packages/solana-intents/src/lib.rs

```diff
@@ -176,3 +177,55 @@ impl<P> From<borsh::io::Error> for IntentError<P> {
         IntentError::DeserializeFailedError(err)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_offchain_message_roundtrip() {
+        let offchain_message = OffchainMessage::new(0, "Fogo Sessions".as_bytes()).unwrap();
+
+        assert_eq!(
+            Message::deserialize(&offchain_message.serialize().unwrap()).unwrap(),
+            Message::OffchainMessage(offchain_message.clone())
+        );
+
+        let mut message_serialized_short: Vec<u8> = offchain_message.serialize().unwrap();
+        message_serialized_short.pop();
```

Comment:
i think these separate test cases, especially when they have different outcomes, should live in different functions

Reply:
doing that would also help to make the test case more readable bc it would be clear why it fails. right now it's not the most readable